File: Arithmetic_&_Basic_Reasoning/Q1. Can Make Arithmetic Progression From Sequence/Q1_CanMakeArithmeticProgressionFromSequence.java
class Solution {
    public boolean canMakeArithmeticProgression(int[] arr) {
        // Step 1: Sort the array
        java.util.Arrays.sort(arr);

        // Step 2: Find the common difference
        int diff = arr[1] - arr[0];

        // Step 3: Check if all consecutive differences are same
        for (int i = 2; i < arr.length; i++) {
            if (arr[i] - arr[i - 1] != diff) {
                return false;   // not an arithmetic progression
            }
        }

        return true; // valid arithmetic progression
    }
}

File: Arithmetic_&_Basic_Reasoning/Q1. Can Make Arithmetic Progression From Sequence/Q1_CanMakeArithmeticProgressionFromSequence.py
class Solution(object):
    def canMakeArithmeticProgression(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """
        
        arr.sort()                     # sort the array to check differences in order
        
        diff = arr[1] - arr[0]         # find the common difference from first two elements
        
        for i in range(2, len(arr)):   # loop through the rest of the elements
            if arr[i] - arr[i - 1] != diff:   # if any difference does not match
                return False           # then it's not an arithmetic progression
        
        return True                    # all differences matched, so return true
